sphinx-docs/screenshot.py

```diff
@@ -0,0 +1,296 @@
+from docutils import nodes
+from docutils.parsers.rst.directives.images import Image
+from docutils.parsers.rst import Directive
+import docutils.parsers.rst.directives as directives
+
+import os
+import uuid
+import json
+
+from exceptions import NotImplementedError
+from errors import ActionError
+from errors import OptionError
+
+from selenium.webdriver.common.keys import Keys
+
+SCREENSHOT_COMMAND = "python ../kalite/manage.py screenshots"
+SCREENSHOT_COMMAND_OPTS = " -v 0 --from-str '%s' --output-dir %s"
+OUTPUT_PATH = os.path.abspath(os.path.join(os.path.dirname(os.path.realpath(__file__)), "images"))
+# These keys are css styles but they need to be camelCased
+FOCUS_CSS_STYLES = { "borderStyle": "solid",
+                     "borderColor": "red",
+                     "borderWidth": "4px",
+                     "borderRadius": "8px",
+                   }
+
+def setup(app):
+    app.add_directive('screenshot', Screenshot)
+    app.connect('env-purge-doc', purge_screenshots)
+    app.connect('env-updated', process_screenshots)
+
+def purge_screenshots(app, env, docname):
+    if not hasattr(env, 'screenshot_all_screenshots'):
+        return
+    env.screenshot_all_screenshots = [s for s in env.screenshot_all_screenshots
+                          if s['docname'] != docname]
+
+def process_screenshots(app, env):
+    if not hasattr(env, 'screenshot_all_screenshots'):
+        return
+
+    all_args = map(lambda x: x['from_str_arg'], env.screenshot_all_screenshots)
+    os.system(SCREENSHOT_COMMAND + SCREENSHOT_COMMAND_OPTS % (json.dumps(all_args), OUTPUT_PATH))
+
+def _parse_focus(arg_str):
+    """ Returns id and annotation after splitting input string.
+
+    First argument should be the jQuery-style selector. An optional 
+    annotation can follow if separated by a separator '|'. Initial
+    whitespace after the '|' will be ignored.
```

Comment:
I think it will be easier to understand these functions if there was a sample argument given in the docstring, showing what format this function is expecting.